core/domain/wipeout_service.py

```diff
@@ -109,11 +178,115 @@ def _delete_user_models(user_id):
     """
     for model_class in models.Registry.get_storage_model_classes(
             [models.NAMES.user]):
-        if (model_class.get_deletion_policy() !=
-                base_models.DELETION_POLICY.KEEP):
+        if (model_class.get_deletion_policy() not in
+                [base_models.DELETION_POLICY.KEEP,
+                 base_models.DELETION_POLICY.NOT_APPLICABLE]):
             model_class.apply_deletion_policy(user_id)
 
 
+def _hard_delete_explorations_and_collections(pending_deletion):
+    """Hard delete the exploration and collection models that are private and
+    solely owned by the user.
+
+    Args:
+        pending_deletion: PendingDeletionRequest.
+    """
+    exp_services.delete_explorations(
+        pending_deletion.user_id,
+        pending_deletion.exploration_ids,
+        force_deletion=True)
+    collection_services.delete_collections(
+        pending_deletion.user_id,
+        pending_deletion.collection_ids,
+        force_deletion=True)
+
+
+def _generate_activity_to_pseudonymized_ids_mapping(activity_ids):
+    """Generate mapping from activity IDs to pseudonymous user IDs.
+
+    Args:
+        activity_ids: list(str). List of activity IDs for which to generate
+            new pseudonymous user IDs. The IDs are of activities that were
+            somehow modified by the user that is currently being deleted.
+
+    Returns:
+        dict(str, str). Mapping between the activity IDs and pseudonymous
+        user IDs. For each activity (with distinct ID) we generate a new
+        pseudonymous user ID.
+    """
+    return {
+        activity_id: user_models.PseudonymizedUserModel.get_new_id()
+        for activity_id in activity_ids
+    }
+
+
+def _delete_story_models(pending_deletion_request):
+    """Pseudonymize the story models for the user with user_id.
+
+    Args:
+        pending_deletion_request: PendingDeletionRequest. The pending deletion
+            request object to be saved in the datastore.
+    """
+    user_id = pending_deletion_request.user_id
+    metadata_models = story_models.StorySnapshotMetadataModel.query(
+        story_models.StorySnapshotMetadataModel.committer_id == user_id
+    ).fetch()
+    story_ids = set([
+        model.get_unversioned_instance_id() for model in metadata_models])
+
+    commit_log_models = story_models.StoryCommitLogEntryModel.query(
+        story_models.StoryCommitLogEntryModel.user_id == user_id
+    ).fetch()
+    commit_log_ids = set(model.story_id for model in commit_log_models)
+    if story_ids != commit_log_ids:
+        logging.warning(
+            'The commit log and snapshot story IDs differ: %s',
+            list(story_ids ^ commit_log_ids))
```

Comment:
Better to print each of these individually. With XOR you can't tell which one came from which list.